Source/Particles/PhysicalParticleContainer.cpp

```diff
@@ -1977,6 +1979,25 @@ PhysicalParticleContainer::getIonizationFunc (const WarpXParIter& pti,
                                 ion_atomic_number);
 }
 
+void PhysicalParticleContainer::resample (Resampling& resampler, const int timestep)
```

Comment:
Would calling this method result in modifications of the `resampler` object? If this is not the case, you can use a `const` reference...

Reply:
Actually it can modify the object (which I don't like too much) because the initialization of the total number of cells in the ResamplingTrigger object is done the first time this method is called. If you think this is better we can do the initialization in a separate method (which we call the first time `mypc->do_resampling()` is called for instance).